src/modules/previewpane/README.md

```diff
@@ -0,0 +1,175 @@
+    [ComVisible(true)]
+    public class XYZPreviewHandler : FileBasedPreviewHandler
+    {
+        private XYZPreviewHandlerControl xyzPreviewHandlerControl;
+
+        /// Call your rendering method here.
+        public override void DoPreview()
+        {
+            this.xyzPreviewHandlerControl.DoPreview(this.FilePath);
+        }
+
+        protected override IPreviewHandlerControl CreatePreviewHandlerControl()
+        {
+            this.xyzPreviewHandlerControl = new xyzPreviewHandlerControl();
+            return this.xyzPreviewHandlerControl;
+        }
+    }
+}
+```
+
+Create a separate Preview Handler Control class and extend the `FormHandlerControl` Class.
+
+```csharp
+using Common;
+
+namespace XYZPreviewHandler
+{
+    public class XYZPreviewHandlerControl : FormHandlerControl
+    {
+        public XYZPreviewHandlerControl()
+        {
+            // ... do your initializations here.
+        }
+
+        public override void DoPreview<T>(T dataSource)
+        {
+            // ... add your preview rendering code here.
+        }
+    }
+}
+```
+
+
+#### Integrate the Preview Handler into PowerToys Settings:
+
+Navigate to the [powerpreview](../previewpane/powerpreview/powerpreview.h) project and edit the `powerpreview.h` file. Add the following Settings Object instance to `m_previewHandlers` settings objects array in the constructor initialization:
+
+```cpp
+// XYZ Preview Handler Settings Object.
+FileExplorerPreviewSettings(
+    false,
+    L"<--YOUR_TOGGLE_CONTROL_ID-->",
+    L"<--A description of your preview handler-->",
+    L"xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx", // your preview handler CLSID.
+    L"<--A display name for your preview handler-->") 
+```
+
+## Install With MSIX
+
+<---To Do--->
+```xml
+
+```
+
+## Managing Preview Handlers
+
+After successful integration, your preview handler should appear in the PowerToys settings UI under the `File Explorer Preview` Tab as shown in Figure 2 below. In here you should be able to enable and disable your preview handler.
+
+<center>
+
+<figure>
+  <img src="../../../doc/images/preview_pane/settings-ui.png" alt="Settings UI - File Explorer Preview Tab" >
+  <figcaption>Figure 2 : Settings UI - File Explorer Preview Tab</figcaption>
+</figure>
+
+</center>
+
+In the general settings of the Settings UI, you should be able to disable and enable all the preview handlers all at once. See Figure 2.
+
+<center>
+
+<figure>
+  <img src="../../../doc/images/preview_pane/general-settings.png" alt="Settings UI - General Settings Tab" >
+  <figcaption>Figure 3 : Settings UI - General Settings Tab</figcaption>
+</figure>
+
+</center>
+
+## Contributing
+
+### Coding Guidance
+
+#### Working With Strings
+
+**YOU SHOULD NOT** have hardcoded strings in your C++ code. Instead, use the following guidelines to add strings to your code. Add the ID of your string to the resource file. XXX must be a unique int in the list (mostly the int ID of the last string id plus one):
```

Comment:
I was just wondering if we should have this section in the README. It is definitely important information, that needs to be pointed out, but isn't it too specific to be put here? If you think this is the best way of letting people know about localization, then it looks good to me.

Reply:
What do you suggest?